### PR TITLE
consul: pythonisms + a couple of required_if clauses

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Minor refactor on the module.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
plugins/modules/clustering/consul/consul.py

### CODE REVIEW

File: plugins/modules/clustering/consul/consul.py [hunk 1]
```diff
@@ -343,63 +343,61 @@ def remove_service(module, service_id):
...
     consulClient.agent.service = PatchedConsulAgentService(consulClient)
     return consulClient
 
 
 def get_service_by_id_or_name(consul_api, service_id_or_name):
     ''' iterate the registered services and find one with the given id '''
-    for name, service in consul_api.agent.services().items():
-        if service['ID'] == service_id_or_name or service['Service'] == service_id_or_name:
+    for dummy, service in consul_api.agent.services().items():
+        if service_id_or_name in [service['ID'], service['Service']]:
```
Comment (felixfontein) [inline plugins/modules/clustering/consul/consul.py:365]:
```suggestion
        if service_id_or_name in (service['ID'], service['Service']):
```

File: plugins/modules/clustering/consul/consul.py [hunk 2]
```diff
@@ -343,63 +343,61 @@ def remove_service(module, service_id):
...
-    if len([p for p in (module.params.get('script'), module.params.get('ttl'), module.params.get('tcp'), module.params.get('http')) if p]) > 1:
+    _checks = [module.params[p] for p in ('script', 'ttl', 'tcp', 'http')]
+
+    if len(c for c in _checks if c) > 1:
         module.fail_json(
             msg='checks are either script, tcp, http or ttl driven, supplying more than one does not make sense')
 
-    if module.params.get('check_id') or module.params.get('script') or module.params.get('ttl') or module.params.get('tcp') or module.params.get('http'):
-
+    if any(_checks):
```
Comment (felixfontein) [inline plugins/modules/clustering/consul/consul.py:376]:
```suggestion
    if module.params['check_id'] or any(_checks):
```

File: plugins/modules/clustering/consul/consul.py [hunk 3]
```diff
@@ -343,63 +343,61 @@ def remove_service(module, service_id):
...
-    for name, service in consul_api.agent.services().items():
-        if service['ID'] == service_id_or_name or service['Service'] == service_id_or_name:
+    for dummy, service in consul_api.agent.services().items():
+        if service_id_or_name in [service['ID'], service['Service']]:
             return ConsulService(loaded=service)
 
 
 def parse_check(module):
-    if len([p for p in (module.params.get('script'), module.params.get('ttl'), module.params.get('tcp'), module.params.get('http')) if p]) > 1:
+    _checks = [module.params[p] for p in ('script', 'ttl', 'tcp', 'http')]
```
Comment (felixfontein) [inline plugins/modules/clustering/consul/consul.py:370]: Why not already collect the ones which evaluate to `True`? In both the next `if` you need exactly these, and not the others.

**Review by felixfontein — COMMENTED** [review]
Thanks! Please add a changelog fragment as well.

File: plugins/modules/clustering/consul/consul.py (hunk 3 above)
Comment (russoz) [reply plugins/modules/clustering/consul/consul.py:370]: yup, makes more sense

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@russoz thanks!

**Pull request merged**